backend/core/src/listings/db/listing-helpers.ts

```diff
@@ -0,0 +1,27 @@
+import { Listing } from "../entities/listing.entity"
+import { ListingsQueryBuilder } from "./listing-query-builder"
+import { Repository } from "typeorm"
+
+export const createQueryBuilder = (
+  listingRepository: Repository<Listing>,
+  alias: string
+): ListingsQueryBuilder => {
+  return new ListingsQueryBuilder(listingRepository.createQueryBuilder(alias))
+}
+
+export const getJurisdictionIdByListingId = async (
+  listingRepository: Repository<Listing>,
+  listingId: string | null
+): Promise<string | null> => {
+  if (!listingId) {
+    return null
+  }
+
+  const listing = await createQueryBuilder(listingRepository, "listings")
+    .where(`listings.id = :listingId`, { listingId })
+    .leftJoin("listings.jurisdiction", "jurisdiction")
+    .select(["listings.id", "jurisdiction.id"])
+    .getOne()
+
+  return listing.jurisdiction.id
+}
```

Comment:
Suggestion (blocking): similar to the user service comments, recommend these get moved into the listing service.